agents_infra/agents_infra/supervisor.py

```diff
@@ -24,14 +25,18 @@ class AgentSupervisor(object):
             exit.
     """
 
-    def __init__(self, agent):
+    def __init__(self, agent, managers=None):
```

Comment:
So the issue I have with how managers are integrated into `AgentSupervisor` is that they don't quite fit in here for now. `AgentSupervisor` is specifically for managing task coroutines and the event loop they're running in. I see two solutions to this:

1. If you think we have to use threads for background servers and not coroutines, have a distinct supervisor managing thread pool.
2. Have background servers run each in their own coroutine asynchronously - then you can build on top of the existing `AgentSupervisor`.

If you're for the first solution, I'd want to know why - why use threads if we already have `shrapnel`?